File: frontend/src/dtos/brokerSettingsDto.ts
import type { FormatOption } from '@/components/SelectFormat.vue'
import type { StructureOption } from '@/components/SelectStructure.vue'

export interface BrokerSettingsDto {
  quickSearches: string[]
  deadLetterQueueSuffix: string
  messageFormat: FormatOption
  messageStructure: StructureOption
}

File: backend/ResQueue/ResQueue/Dtos/Broker/BrokerSettingsDto.cs
namespace ResQueue.Dtos;

public record BrokerSettingsDto(
    List<string> QuickSearches,
    string DeadLetterQueueSuffix,
    string MessageFormat,
    string MessageStructure,
    string QueueTrimPrefix,
    string? DefaultQueueSortField,
    int DefaultQueueSortOrder,
    string DefaultQueueSearch
);